CMakeLists.txt

```diff
@@ -63,6 +63,14 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" OR "${CMAKE_CXX_SIMULATE_ID}" STR
   add_compile_options(/D_CRT_SECURE_NO_WARNINGS) # TODO: /Wall /WX
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
   add_compile_options(-Wall -Werror)
+  if ("${CMAKE_SYSTEM_NAME}" MATCHES "FreeBSD|OpenBSD")
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+      add_compile_options(-Wno-delete-non-virtual-dtor)
+      add_compile_options(-Wno-self-assign-field)
+    else ()
+      add_compile_options(-Wno-cpp)
+    endif ()
+  endif ()
```

Comment:
I'm ok with adding these but do they actually generate a lot of errors? Would it be worth just fixing the warnings/errors instead of ignoring them?

Reply:
build error message:
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::copy::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copy::Driver, std::__1::allocator<builtin::copy::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copy::Driver, std::__1::allocator<builtin::copy::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::copy::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:59:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::copy::Driver>' requested here
        std::make_shared<builtin::copy::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::copyPlist::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copyPlist::Driver, std::__1::allocator<builtin::copyPlist::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copyPlist::Driver, std::__1::allocator<builtin::copyPlist::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::copyPlist::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:60:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::copyPlist::Driver>' requested here
        std::make_shared<builtin::copyPlist::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::copyStrings::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copyStrings::Driver, std::__1::allocator<builtin::copyStrings::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copyStrings::Driver, std::__1::allocator<builtin::copyStrings::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::copyStrings::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:61:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::copyStrings::Driver>' requested here
        std::make_shared<builtin::copyStrings::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::copyTiff::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copyTiff::Driver, std::__1::allocator<builtin::copyTiff::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::copyTiff::Driver, std::__1::allocator<builtin::copyTiff::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::copyTiff::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:62:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::copyTiff::Driver>' requested here
        std::make_shared<builtin::copyTiff::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::infoPlistUtility::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::infoPlistUtility::Driver, std::__1::allocator<builtin::infoPlistUtility::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::infoPlistUtility::Driver, std::__1::allocator<builtin::infoPlistUtility::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::infoPlistUtility::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:63:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::infoPlistUtility::Driver>' requested here
        std::make_shared<builtin::infoPlistUtility::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::lsRegisterURL::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::lsRegisterURL::Driver, std::__1::allocator<builtin::lsRegisterURL::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::lsRegisterURL::Driver, std::__1::allocator<builtin::lsRegisterURL::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::lsRegisterURL::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:64:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::lsRegisterURL::Driver>' requested here
        std::make_shared<builtin::lsRegisterURL::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::productPackagingUtility::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::productPackagingUtility::Driver, std::__1::allocator<builtin::productPackagingUtility::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::productPackagingUtility::Driver, std::__1::allocator<builtin::productPackagingUtility::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::productPackagingUtility::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:65:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::productPackagingUtility::Driver>' requested here
        std::make_shared<builtin::productPackagingUtility::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::validationUtility::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::validationUtility::Driver, std::__1::allocator<builtin::validationUtility::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::validationUtility::Driver, std::__1::allocator<builtin::validationUtility::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::validationUtility::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:66:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::validationUtility::Driver>' requested here
        std::make_shared<builtin::validationUtility::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
/usr/include/c++/v1/memory:3656:5: error: destructor called on non-final 'builtin::embeddedBinaryValidationUtility::Driver' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/usr/include/c++/v1/memory:3612:5: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::embeddedBinaryValidationUtility::Driver, std::__1::allocator<builtin::embeddedBinaryValidationUtility::Driver> >::__on_zero_shared' requested here
    __shared_ptr_emplace(_Alloc __a)
    ^
/usr/include/c++/v1/memory:4277:26: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<builtin::embeddedBinaryValidationUtility::Driver, std::__1::allocator<builtin::embeddedBinaryValidationUtility::Driver> >::__shared_ptr_emplace' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/usr/include/c++/v1/memory:4656:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<builtin::embeddedBinaryValidationUtility::Driver>::make_shared<>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
../Libraries/builtin/Sources/Registry.cpp:67:14: note: in instantiation of function template specialization 'std::__1::make_shared<builtin::embeddedBinaryValidationUtility::Driver>' requested here
        std::make_shared<builtin::embeddedBinaryValidationUtility::Driver>(),
             ^
/usr/include/c++/v1/memory:3656:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
                      ^
9 errors generated.
[69/537] Building CXX object Libraries...tin.dir/Sources/copyPlist/Driver.cpp.o
ninja: build stopped: subcommand failed.
gmake: *** [Makefile:21: all] Error 1

Reply:
gcc build error mssage:
[332/537] Building CXX object Librarie...ocess.dir/Sources/DefaultContext.cpp.o
FAILED: Libraries/process/CMakeFiles/process.dir/Sources/DefaultContext.cpp.o 
/usr/local/bin/g++48  -Dprocess_EXPORTS -I../Libraries/process/Headers -I../Libraries/ext/Headers -I../Libraries/libutil/Headers -std=c++11 -fno-rtti -fno-exceptions -fPIC   -Wall -Werror -Wno-multichar -Wno-sign-compare -D_GLIBCXX_USE_C99 -D_GLIBCXX_USE_C99_MATH -D_GLIBCXX_USE_C99_MATH_TR1 -D_GLIBCXX_USE_CXX11_ABI=1 -MD -MT Libraries/process/CMakeFiles/process.dir/Sources/DefaultContext.cpp.o -MF Libraries/process/CMakeFiles/process.dir/Sources/DefaultContext.cpp.o.d -o Libraries/process/CMakeFiles/process.dir/Sources/DefaultContext.cpp.o -c ../Libraries/process/Sources/DefaultContext.cpp
In file included from ../Libraries/process/Sources/DefaultContext.cpp:40:0:
/usr/include/sys/syslimits.h:43:2: error: #warning "No user-serviceable parts inside." [-Werror=cpp]
 #warning "No user-serviceable parts inside."
  ^
cc1plus: all warnings being treated as errors

Reply:
FAILED: Libraries/pbxspec/CMakeFiles/pbxspec.dir/Sources/PBX/FileType.cpp.o 
/usr/bin/c++  -Dpbxspec_EXPORTS -I../Libraries/pbxspec/Headers -I../Libraries/pbxspec/PrivateHeaders -I../Libraries/pbxsetting/Headers -I../Libraries/process/Headers -I../Libraries/ext/Headers -I../Libraries/libutil/Headers -I../Libraries/plist/Headers -std=c++11 -fno-rtti -fno-exceptions -fPIC   -Wall -Werror -Wno-delete-non-virtual-dtor -fcolor-diagnostics -MD -MT Libraries/pbxspec/CMakeFiles/pbxspec.dir/Sources/PBX/FileType.cpp.o -MF Libraries/pbxspec/CMakeFiles/pbxspec.dir/Sources/PBX/FileType.cpp.o.d -o Libraries/pbxspec/CMakeFiles/pbxspec.dir/Sources/PBX/FileType.cpp.o -c ../Libraries/pbxspec/Sources/PBX/FileType.cpp
../Libraries/pbxspec/Sources/PBX/FileType.cpp:51:46: error: assigning field to itself [-Werror,-Wself-assign-field]
    _extensions                              = _extensions; /* Explicitly not inherited. */
                                             ^
1 error generated.
[109/537] Building CXX object Librarie...ec.dir/Sources/PBX/Specification.cpp.o
ninja: build stopped: subcommand failed.

Reply:
It's probably safer to add virtual destructors to these classes if they have virtual functions, but like I said, I'm ok with ignoring these warnings for now. Up to you if you'd like to improve this.

Reply:
I'm fixed the warings!